pkg/deploy/spindeploy/transformer/telemety.go

```diff
@@ -0,0 +1,114 @@
+package transformer
+
+import (
+	"context"
+	"github.com/armory/spinnaker-operator/pkg/apis/spinnaker/v1alpha2"
+	"github.com/armory/spinnaker-operator/pkg/generated"
+	"github.com/armory/spinnaker-operator/pkg/inspect"
+	"github.com/armory/spinnaker-operator/version"
+	"github.com/ghodss/yaml"
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	EchoConfigFile     = "echo.yml"
+	KubernetesOperator = "kubernetes_operator"
+	TelemetryKey       = "telemetry"
+)
+
+type telemetryTransformer struct {
+	log logr.Logger
+}
+
+type telemetryTransformerGenerator struct{}
+
+type DeploymentMethod struct {
+	DeploymentType    string `json:"type"`
+	DeploymentVersion string `json:"version"`
+}
+
+type Telemetry struct {
+	Enabled                 bool             `json:"enabled"`
+	Endpoint                string           `json:"endpoint"`
+	InstanceId              string           `json:"instanceId"`
+	SpinnakerVersion        string           `json:"spinnakerVersion"`
+	DeploymentMethod        DeploymentMethod `json:"deploymentMethod"`
+	ConnectionTimeoutMillis int              `json:"connectionTimeoutMillis"`
+	ReadTimeoutMillis       int              `json:"readTimeoutMillis"`
+}
+
+func (t *telemetryTransformerGenerator) NewTransformer(svc v1alpha2.SpinnakerServiceInterface,
+	client client.Client, log logr.Logger) (Transformer, error) {
+	tr := telemetryTransformer{log: log}
+	return &tr, nil
+}
+
+func (t *telemetryTransformerGenerator) GetName() string {
+	return "Telemetry"
+}
+
+func (t *telemetryTransformer) TransformConfig(ctx context.Context) error {
+	return nil
+}
+
+func (t *telemetryTransformer) TransformManifests(ctx context.Context, scheme *runtime.Scheme, gen *generated.SpinnakerGeneratedConfig) error {
+	for _, cfg := range gen.Config {
+		for k := range cfg.Resources {
```

Comment:
I think it would be better to explicitly search for `echo` service config here, and once found iterate only echo's Resources instead of iterating the resources of all services:
```
for svc, cfg := range gen.Config {
    if svc != "echo" {
        continue
    }
    for k := range cfg.Resources {
    ...
```